crowbar_framework/app/models/service_object.rb

```diff
@@ -854,6 +854,56 @@ def violates_admin_constraint?(elements, role, nodes_is_admin = {})
     false
   end
 
+  def violates_platform_constraint?(elements, role)
+    if role_constraints[role]
+      elements[role].each do |element|
+        next if is_cluster? element
+        node = NodeObject.find_node_by_name(element)
+
+        (platform, version) = role_constraints[role]["platform"]
+        platform = {
+          :platform => platform,
+          :platform_version => version
+        }
+        platform.any? do |key, value|
```

Comment:
I think you need to explictly return the result of this (also platform.all? in the second function)